testutils/src/main/scala/za/co/absa/enceladus/testutils/e2eSparkRunner/E2ESparkRunner.scala

```diff
@@ -0,0 +1,131 @@
+  private final val compClass = "--class za.co.absa.enceladus.testutils.datasetComparison.ComparisonJob"
+
+  private def getComparisonConf(newPath: String, refPath: String, outPath: String, keys: String): String = {
+    s"--raw-format parquet --new-path $newPath --ref-path $refPath --out-path $outPath --keys $keys"
+  }
+
+  private def getStandartizationPath(cmd: CmdConfig): String = {
+    "conformance-output/standardized-" +
+      s"${cmd.datasetName.split(" ")(1)}-" +
+      s"${cmd.datasetVersion.split(" ")(1)}-" +
+      s"${cmd.reportDate.split(" ")(1)}-" +
+      s"${cmd.reportVersion.split(" ")(1)}"
+  }
+
+  private def getConformancePath(cmd: CmdConfig): String = {
+    s"${cmd.datasetName.split(" ")(1)}/" +
+      s"enceladus_info_date=${cmd.reportDate.split(" ")(1)}/" +
+      s"enceladus_info_version=${cmd.reportVersion.split(" ")(1)}"
+  }
+
+  private def getConformanceJobConf(cmd: CmdConfig): String = {
+    s"${cmd.datasetName} ${cmd.datasetVersion} ${cmd.reportDate} ${cmd.reportVersion} ${cmd.menasCredentialsFile}"
+  }
+
+  private def getStandartizationJobConf(cmd: CmdConfig): String = {
+    val base: String = s"${cmd.datasetName} " +
+      s"${cmd.datasetVersion} " +
+      s"${cmd.reportDate} " +
+      s"${cmd.reportVersion} " +
+      s"${cmd.menasCredentialsFile}"
+
+    val format: String = cmd.rawFormat match {
+      case "csv" => s"--raw-format ${cmd.rawFormat} --delimiter ${cmd.csvDelimiter.get} --header ${cmd.csvHeader.get}"
+      case "xml" => s"--raw-format ${cmd.rawFormat} --row-tag ${cmd.rowTag.get}"
+      case _ => s"--raw-format ${cmd.rawFormat}"
+    }
+
+    s"$base $format"
+  }
+
+  private def runBashCmd(bashCmd: String): String = {
+    (s"echo $bashCmd" #| "bash").!!
+  }
+
+  def main(args: Array[String]): Unit = {
+    val cmd = CmdConfig.getCmdLineArguments(args)
+
+    val sparkConf = Source.fromFile(cmd.sparkConfFile).getLines().mkString
+
+    val stdJobConf = getStandartizationJobConf(cmd)
+    val confJobConf = getConformanceJobConf(cmd)
+
+    val stdPaths = getStandartizationPath(cmd)
+    val confPaths = getConformancePath(cmd)
+
+    val defaultStdOut = s"/tmp/$stdPaths"
+    val defaultConfOut = s"/publish/$confPaths"
+
+    val refStdOut = s"/ref/tmp/$stdPaths"
+    val refConfOut = s"/ref/publish/$confPaths"
+
+    val cmpStdPath = s"/cmp/tmp/$stdPaths"
+    val cmpConfPath = s"/cmp/publish/$confPaths"
+
+    val stdComparisonConf = getComparisonConf(defaultStdOut, refStdOut, cmpStdPath, cmd.keys)
+    val confComparisonConf = getComparisonConf(defaultConfOut, refConfOut, cmpConfPath, cmd.keys)
+
+    val standartisation = s"spark-submit $sparkConf $stdClass $stdJarPath $stdJobConf"
+    log.debug(standartisation)
+    log.info("Running Standartization")
+    lazy val standartisationResult: String = runBashCmd(standartisation)
+    val stdTime = HelperFunctions.calculateTime { standartisationResult }
```

Comment:
Make it independent from the fact that the `standartisationResult` is defined as `lazy`.

Reply:
reworked. Now using unapply on a case class

Reply:
Nice ⚡️